apps/api/src/auth/services/auth.interceptor.ts

```diff
@@ -9,41 +12,67 @@ import { kvStore } from "@src/middlewares/userMiddleware";
 import { UserOutput, UserRepository } from "@src/user/repositories";
 import { env } from "@src/utils/env";
 import { getJwks, useKVStore, verify } from "@src/verify-rsa-jwt-cloudflare-worker-main";
+import { ApiKeyAuthService } from "./api-key/api-key-auth.service";
 
 @singleton()
 export class AuthInterceptor implements HonoInterceptor {
+  private readonly logger = LoggerService.forContext(AuthInterceptor.name);
+
   constructor(
     private readonly abilityService: AbilityService,
     private readonly userRepository: UserRepository,
     private readonly authService: AuthService,
-    private readonly anonymousUserAuthService: AuthTokenService
+    private readonly anonymousUserAuthService: AuthTokenService,
+    private readonly apiKeyAuthService: ApiKeyAuthService
   ) {}
 
   intercept() {
     return async (c: Context, next: Next) => {
-      const bearer = c.req.header("authorization");
+      try {
+        const bearer = c.req.header("authorization");
 
-      const anonymousUserId = bearer && (await this.anonymousUserAuthService.getValidUserId(bearer));
+        const anonymousUserId = bearer && (await this.anonymousUserAuthService.getValidUserId(bearer));
 
-      if (anonymousUserId) {
-        const currentUser = await this.userRepository.findAnonymousById(anonymousUserId);
-        await this.auth(currentUser);
-        c.set("user", currentUser);
-        return await next();
-      }
+        if (anonymousUserId) {
+          const currentUser = await this.userRepository.findAnonymousById(anonymousUserId);
+          assert(currentUser, 401, "Invalid anonymous user");
+          await this.auth(currentUser);
+          c.set("user", currentUser);
+          return await next();
+        }
+
+        const userId = bearer && (await this.getValidUserId(bearer, c));
+
+        if (userId) {
+          const currentUser = await this.userRepository.findByUserId(userId);
+          assert(currentUser, 401, "Invalid user");
+          await this.auth(currentUser);
+          c.set("user", currentUser);
+          return await next();
+        }
 
-      const userId = bearer && (await this.getValidUserId(bearer, c));
+        const apiKey = c.req.header("x-api-key");
 
-      if (userId) {
-        const currentUser = await this.userRepository.findByUserId(userId);
-        await this.auth(currentUser);
-        c.set("user", currentUser);
+        if (apiKey) {
+          const apiKeyOutput = await this.apiKeyAuthService.getAndValidateApiKeyFromHeader(apiKey);
+          const currentUser = await this.userRepository.findByUserId(apiKeyOutput.userId);
+          assert(currentUser, 401, "Invalid API key user");
+          await this.auth(currentUser);
+          c.set("user", currentUser);
+          return await next();
+        }
+
+        this.authService.ability = this.abilityService.EMPTY_ABILITY;
         return await next();
-      }
+      } catch (error) {
+        if (error instanceof HTTPException) {
```

Comment:
what's the purpose of this? `assert` already throws distinguishable errors with codes/names/etc. These errors are handled in global error handler

Reply:
You're right I'll remove it.